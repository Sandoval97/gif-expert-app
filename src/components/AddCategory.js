import React, { useState } from 'react'
import PropTypes from 'prop-types';

function AddCategory({ setCategories }) {
    const [valueInput, setValue] = useState('')

    const onSubmitEvent = (e) => {
        e.preventDefault()

        if (valueInput.trim().length > 2) {
            setCategories(categories => [valueInput, ...categories])
            setValue('')
        }
    }

    return (
        <form onSubmit={onSubmitEvent}>
            <input
                value={valueInput}
                onChange={(e) => setValue(e.target.value)}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

export default AddCategory;