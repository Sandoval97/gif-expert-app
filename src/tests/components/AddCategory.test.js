import React from "react";
import { shallow } from "enzyme"
import AddCategory from "../../components/AddCategory"

describe("Pruebas en componente AddCategory", () => {
    const setCategories = jest.fn()
    var wrapper;

    beforeEach(() => {
        wrapper = shallow(<AddCategory setCategories={setCategories} />)
    })

    test("El componente no debe cambiar su estructura tipica en JSX", () => {
        expect(wrapper).toMatchSnapshot()
    })

    test("Se debio haber llamado el metodo setCategories", () => {
        const input = wrapper.find('input')
        const value = 'Hello World'
        input.simulate('change', {
            target: { value }
        })

        const form = wrapper.find('form')
        form.simulate('submit', {
            preventDefault: jest.fn()
        })
        expect(setCategories).toHaveBeenCalled();
    })


    test("No se debio haber llamado el metodo setCategories", () => {

        const form = wrapper.find('form')
        form.simulate('submit', {
            preventDefault: jest.fn()
        })
        expect(setCategories).not.toHaveBeenCalled();
    })

    test("Luego de todo el flujo se debe haber limpiado el input", () => {
        const input = wrapper.find('input')
        const value = 'Hello World!'
        input.simulate('change', {
            target: { value }
        })

        const form = wrapper.find('form')
        form.simulate('submit', {
            preventDefault: jest.fn()
        })

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(input.prop('value')).toBe('');
    })


})