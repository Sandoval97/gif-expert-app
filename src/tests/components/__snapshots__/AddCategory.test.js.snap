// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente AddCategory El componente no debe cambiar su estructura tipica en JSX 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    value=""
  />
</form>
`;
